### PR TITLE
Do not promote fp8 operands until GemmAutotuning has been done.

### PR DESCRIPTION
**Pull request by elfiegg**
Do not promote fp8 operands until GemmAutotuning has been done.
The PR doesn't change the current behavior as all fp8 gemms are currently processed by GemmRewriter(translated into cublasLT calls) and will not reach GemmAutotuner. 
We're planning to let fp8 gemm fusions go through the autotuner.

### CODE REVIEW

File: xla/service/gpu/gpu_compiler.cc [hunk 1]
```diff
@@ -1369,15 +1369,18 @@ absl::Status GpuCompiler::OptimizeHloPostLayoutAssignment(
   const FloatSupport f8e4m3b11fnuz_support(F8E4M3B11FNUZ, F16);
   const GpuFloatSupport f8e5m2fnuz_support(gpu_version, F8E5M2FNUZ, F16);
   const GpuFloatSupport f8e4m3fnuz_support(gpu_version, F8E4M3FNUZ, F16);
-  auto add_float_normalization = [&](HloPassPipeline& pipeline) {
+  auto add_float_normalization = [&](HloPassPipeline& pipeline,
+                                     bool fp8_support = false) {
```
Comment (tdanyluk) [inline xla/service/gpu/gpu_compiler.cc:1373]: Maybe change "bool fp8_support" to "bool promote_fp8", and remove the default value, to make it more explicit.

**Review by tdanyluk — COMMENTED** [review]
Thanks.
I think it's fine as long as it really doesn't regress anything.
I just had one short comment.

**Comment by mooskagh** [comment]
Looks good, but note that I'm currently enabling the fp8 fusions to go through Triton, meaning that:
* `add_float_normalization` won't normalize fp8 in any case.
* `GemmRewriter` won't have `fp8` param, it will do both rewrites at the same time (first time -- after `GemmFusion`)
* `GemmFusion` will fuse fp8 gemms.

**Comment by mooskagh** [comment]
Thinking about it a bit more, I'm struggling a bit to understand what is the intended plan.

If we normalize fp8 in the first call but not in the second, what would be the example flow where it makes a difference?

**Pull request closed without merging**